sw/device/examples/sram_program/BUILD

```diff
@@ -31,6 +32,48 @@ opentitan_ram_binary(
         "//hw/top_earlgrey/sw/autogen:top_earlgrey",
         "//sw/device/lib/base:macros",
         "//sw/device/lib/runtime:log",
+        "//sw/device/lib/testing:pinmux_testutils",
         "//sw/device/lib/testing/test_framework:check",
     ],
 )
+
+opentitan_gdb_fpga_cw310_test(
+    name = "sram_program_fpga_cw310_test",
+    exit_success_pattern = "sram_program\\.c:47\\] PC: 0x1000208c, SRAM: \\[0x10000000, 0x10020000\\)",
+    gdb_script = """
+        target extended-remote :3333
+        set pagination off
+
+        echo :::: Send OpenOCD the 'reset halt' command.\\n
+        monitor reset halt
+
+        echo :::: Set RAM as RWX by executing code between _start and test_rom_main()\\n
+        file test_rom.elf
+        break rom_test_main
+        continue
+        delete breakpoints
+
+        echo :::: Load the SRAM program onto the device and check integrity.\\n
+        file sram_program.elf
+        load sram_program.elf
+        compare-sections
+
+        info registers
+        echo :::: Update registers before calling sram_main().\\n
+        set $sp = _stack_end
+        set $gp = __global_pointer$
+        info registers
```

Comment:
I know we talked about this already but once we start running this with ROM, we should initialize the ePMP here as well.

Reply:
Noted!

Is it acceptable to reuse code from the ROM the same way we reused the test ROM[^0] ? I had some trouble reproducing those instructions (e.g. `csrw pmpcfg3, t0`) in GDB without jumping to that part of the test ROM.

I believe the relevant PMP CSRs are Ibex-specific[^1] (e.g. `pmpcfg3`), which is why I can't just ask GDB to `set $pmpcfg3 = $t0`. However, we may be able to lean on OpenOCD's `reg` command[^2] via the `monitor` command in GDB. I haven't tried this yet, but I wonder whether we can do `monitor reg 0x3a3 0x9f000000` to emulate `li t0, (0x9f << 24); csrw pmpcfg3, t0`.

[^0]: https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/testing/test_rom/test_rom_start.S;l=205-211;drc=a836c21da9116be160a50553ac41b645c8b81a33
[^1]: https://ibex-core.readthedocs.io/en/latest/03_reference/cs_registers.html
[^2]: https://openocd.org/doc/html/General-Commands.html

Reply:
Hmm, for tests, sure but during manufacturing we need to be able to configure those registers. Thanks for the investigation and references. We can take a look at this together next week.